Recreate #15175

icecream17 opened the pull request:
See #15175
Resolves #15172 (although it's already closed anyways)

Commit:
Recreate #15175

Commit:
fix lint

sadick254 reviewed (commented):
The changes look good. Do we know what events that lead to the error?

icecream17 commented:
From what I gather about #15172 we don't know why the error happened.

sadick254 commented:
I think a good approach would be to display the error as a notification. This way we can have visibility when the error occurs. A user will be able to create an issue from the notification modal. See https://flight-manual.atom.io/api/v1.57.0/NotificationManager/

Commit:
Notify user on error

I'm assuming that `addError` displays an error.

I'm also logging the error on the console - for theoretical debugging in case the notification is dismissed - but maybe it's unnecessary.
I renamed `event` to `error` in the `onupgradeneeded` handler

icecream17 commented:
Is the console.error unnecessary now that there's a notification?

sadick254 commented:
The `console.error` is fine.

UziTech commented:
I think [`addFatalError`](https://flight-manual.atom.io/api/v1.57.0/NotificationManager/#instance-addFatalError) might be better so the button to easily create an issue appears.

sadick254 commented on src/state-store.js:
```diff
@@ -13,13 +13,24 @@ module.exports = class StateStore {
         dbOpenRequest.onupgradeneeded = event => {
           let db = event.target.result;
+          db.onerror = error => {
+            atom.notifications.addError('Error loading database', {
```
```suggestion
            atom.notifications.addFatalError('Error loading database', {
```

sadick254 commented on src/state-store.js:
```diff
@@ -13,13 +13,24 @@ module.exports = class StateStore {
           resolve(dbOpenRequest.result);
         };
         dbOpenRequest.onerror = error => {
+          atom.notifications.addError('Could not connect to indexedDB', {
```
```suggestion
          atom.notifications.addFatalError('Could not connect to indexedDB', {
```

sadick254 reviewed (changes requested):
I agree with @UziTech

Commit:
change to addFatalError (1) [skip ci]

Co-authored-by: Sadick <sadickjunior@gmail.com>

Commit:
change to addFatalError (2)

Co-authored-by: Sadick <sadickjunior@gmail.com>
